src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -153,6 +167,24 @@ protected void tick(@Nonnull Block b) {
                         nextFluid.setType(Material.AIR);
                     }
 
+                    return;
+                }
+                if (SlimefunUtils.isItemSimilar(menu.getItemInSlot(slot), emptyBottle, true, false)) {
+                    ItemStack glassBottle = getFilledBottle(fluid);
+
+                    if (!menu.fits(glassBottle, getOutputSlots())) {
+                        return;
+                    }
+
+                    Block nextFluid = findNextFluid(fluid);
+
+                    if (nextFluid != null) {
+                        removeCharge(b.getLocation(), ENERGY_CONSUMPTION);
+                        menu.consumeItem(slot);
+                        menu.pushItem(glassBottle, getOutputSlots());
+                        nextFluid.setType(Material.AIR);
```

Comment:
This is an interesting one, and would need input from others, but water bottles don't consume the water block they are used on when used in a vanilla context, should it here? Alternatively a Cauldron will lose 1/3 a bucket of water when filling up a bottle so perhaps it would be nice to add a 1 in 3 chance to consume the water, just to add a little interesting quirk into the machine? Debatable :D

Reply:
agree on that, no need to debate for it I guess since its a vanilla mechanics xD later on players might just complain from what I did, will just add a low chance